docs/examples/gnoi/golang/index.md

```diff
@@ -0,0 +1,141 @@
+---
+layout: default
+title: "gNOI golang examples"
+date: 2021-07-19 09:00:00 --0600
+categories:
+---
+
+## About gRPC
+
+gRPC - Google Remote Procedure Call
+
+gRPC uses protobuf and HTTP/2
+
+## About gNOI
+
+gNOI - gRPC Network Operations Interface
+
+gNOI defines a set of gRPC-based microservices for executing operational commands on network devices.
+
+gNOI [github repository](https://github.com/openconfig/gnoi)
+
+As example, we use the [system gNOI proto file](https://github.com/openconfig/gnoi/blob/master/system/system.proto) and a example from the [os gNOI proto file](https://github.com/openconfig/gnoi/tree/master/os)
+
+- Ping executes the ping command on the target and streams back the results
+- Traceroute executes the traceroute command on the target and streams back the results
+- As you can see in the proto file, the field VRF is not defined for these messages
+- OS will quickly tell the operator what version of the OS installed as well as the current supervisor which is active.  cEOS does not have a supervisor. 
+
+## About gNOI support on EOS
+
+Please refer to this [link](https://eos.arista.com/eos-4-24-2f/gnoi/) for the gNOI support by EOS.
+
+Examples:
+
+- [gnoi ping](https://eos.arista.com/eos-4-22-1f/gnoi-ping/)
+- [gnoi traceroute](https://eos.arista.com/eos-4-22-1f/gnoi-traceroute/)
+
+## golang overview 
+
+Current version of golang running on my system is as follows.
+
+```shell
+go version go1.16.5 linux/amd64
+```
+
+## change directory into golang gNOI examples.
+
+```shell
+cd src/gnoi-golang/
+```
+
+## File structure 
+
+```
+.
+├── go.mod
+├── go.sum
+├── os-services
+│   └── verify.go
+└── system-services
+    ├── pinggnoi.go
+    ├── rebootgnoi.go
+    ├── timegnoi.go
+    └── traceroutegnoi.go
+```
+
+The os-services contain the [OS proto services](https://github.com/openconfig/gnoi/tree/master/os) and the [system-services](https://github.com/openconfig/gnoi/tree/master/system) contain the system-services protos services.  Time and reboot are not currently supported but provide examples.
+
+## Device config
+
+```shell
+interface Management1
+   description oob_management
+   vrf MGMT
+   ip address 10.73.1.118/24
+
+username admin secret 0 admin
+
+management api gnmi
+   transport grpc def
+```
+
+```shell
+ceos1#show management api gnmi
+Enabled:            Yes
+Server:             running on port 6030
+SSL Profile:        none
+QoS DSCP:           none
+ceos1#
+```
```

Comment:
this can be removed, instead reference existing configuration elements in the site.